src/core/sagas/recommendations.ts

```diff
@@ -49,56 +59,110 @@ export namespace Tasks {
 
         yield effects.put(flux.actions.receiveRecommendationsProducts(SearchAdapter.augmentProducts(results)));
       }
-      return [];
     } catch (e) {
       yield effects.put(flux.actions.receiveRecommendationsProducts(e));
     }
   }
 
-  export function* fetchPastPurchases(flux: FluxCapacitor, { payload }: Actions.FetchPastPurchases) {
+  export function* fetchSkus(config: Configuration, endpoint: string, query?: string) {
+    const securedPayload = ConfigAdapter.extractSecuredPayload(config);
+    if (securedPayload) {
+      const url = `https://${config.customerId}.groupbycloud.com/orders/v1/public/skus/${endpoint}`;
+      const response = yield effects.call(utils.fetch, url, Adapter.buildBody({
+        securedPayload,
+        query
+      }));
+      return yield response.json();
+    }
+    throw new MissingPayload();
+  }
+
+  // tslint:disable-next-line max-line-length
+  export function* fetchProductsFromSkus(flux: FluxCapacitor, skus: Store.PastPurchases.PastPurchaseProduct[], request: Request) {
+    const ids: string[] = skus.map(({ sku }) => sku);
+    return yield effects.call(
+      [flux.clients.bridge, flux.clients.bridge.search],
+      {
+        ...request,
+        biasing: <Biasing>{
+          restrictToIds: ids,
+        },
+        sort: <Sort[]>[{ type: 'ByIds', ids }],
+      });
+  }
+
+  export function* fetchPastPurchases(flux: FluxCapacitor, action: Actions.FetchPastPurchases) {
     try {
-      const config = yield effects.select(Selectors.config);
-      const productCount = config.recommendations.pastPurchases.productCount;
+      const config: Configuration = yield effects.select(Selectors.config);
+      const productCount = ConfigAdapter.extractPastPurchaseProductCount(config);
       if (productCount > 0) {
-        const url = `http://${config.customerId}.groupbycloud.com/orders/public/skus/_search`;
-        // TODO: change to be the real/right request
-        const response = yield effects.call(fetch, url, Adapter.buildBody(<any>{
-          // slide 9 of Past Purchase Epic?
-          keyword: payload && undefined
-        }));
-        const result = yield response.json();
-        // TODO: modify data so it's in the right form?
-        yield effects.put(flux.actions.receivePastPurchases(result.result));
+        const { result } = yield effects.call(fetchSkus, config, 'popular');
+        yield effects.put(flux.actions.receivePastPurchaseSkus(result));
+        yield effects.put(flux.actions.fetchPastPurchaseNavigations());
+      } else {
+        yield effects.put(flux.actions.receivePastPurchaseSkus([]));
       }
-      return [];
     } catch (e) {
-      return effects.put(flux.actions.receivePastPurchases(e));
+      if (!(e instanceof MissingPayload)) { // pass through misisng payloads
+        return effects.put(flux.actions.receivePastPurchaseSkus(e));
+      }
```

Comment:
what happens if it is an instance of MissingPayload?

Reply:
It gets ignored. I'm passing through missing payload errors and simply allowing the action to do nothing since there's no need to roll back the state.